packages/@vue/cli-plugin-typescript/generator/convert.js

```diff
@@ -1,22 +1,30 @@
-module.exports = (api, { tsLint = false } = {}) => {
-  // delete all js files that have a ts file of the same name
-  // and simply rename other js files to ts
+module.exports = (api, { tsLint = false } = {}, convertAllFiles = true) => {
   const jsRE = /\.js$/
   const excludeRE = /^tests\/e2e\/|(\.config|rc)\.js$/
   const convertLintFlags = require('../lib/convertLintFlags')
   api.postProcessFiles(files => {
-    for (const file in files) {
-      if (jsRE.test(file) && !excludeRE.test(file)) {
-        const tsFile = file.replace(jsRE, '.ts')
-        if (!files[tsFile]) {
-          let content = files[file]
-          if (tsLint) {
-            content = convertLintFlags(content)
+    if (convertAllFiles) {
+      // delete all js files that have a ts file of the same name
+      // and simply rename other js files to ts
+      for (const file in files) {
+        if (jsRE.test(file) && !excludeRE.test(file)) {
+          const tsFile = file.replace(jsRE, '.ts')
+          if (!files[tsFile]) {
+            let content = files[file]
+            if (tsLint) {
+              content = convertLintFlags(content)
+            }
+            files[tsFile] = content
           }
-          files[tsFile] = content
+          delete files[file]
         }
-        delete files[file]
       }
+    } else {
+      // rename only main file to main.ts
+      const content = files[api.entryFile]
+      const tsFile = api.entryFile.replace(jsRE, '.ts')
```

Comment:
I might be completely wrong here but don't we need to `convertLintFlags` here too?

Reply:
Yes, you're right 😅 Added a `convertLintFlags`